Documentation/contributing/testing/ci.rst

```diff
@@ -37,41 +37,41 @@ After you don't need to run tests on your branch, please remove the branch from
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sDemos``               | Runs all k8s demo tests                   |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sKubeProxyFreeMatrix`` | Runs all k8s kube-proxy free matrix tests |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sFQDNTest``            | Runs all k8s fqdn tests                   |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sHealthTest``          | Runs all k8s health tests                 |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sHubbleTest``          | Runs all k8s Hubble tests                 |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sIdentity``            | Runs all k8s identity tests               |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sIstioTest``           | Runs all k8s Istio tests                  |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sKafkaPolicyTest``     | Runs all k8s Kafka tests                  |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sPolicyTest``          | Runs all k8s policy tests                 |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sServicesTest``        | Runs all k8s services tests               |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus K8sUpdates``             | Runs k8s update tests                     |
-   +---------------------------------------+-------------------------------------------+
-   | ``test-focus Runtime``                | Runs all runtime tests                    |
-   +---------------------------------------+-------------------------------------------+
+   It is also possible to run specific tests from this suite via ``test-only``. The comment can contain 3 arguments: ``--focus`` which specifies which tests should be run, ``--kernel_version`` for supported kernel version (net-next, 49, 419 are possible values right now), ``--k8s_version`` for k8s version. If you want to run only one ``It`` block, you need to prepend it with a test suite and create a regex, e.g ``test-only --focus="K8sDatapathConfig.*Check connectivity with automatic direct nodes routes" --k8s_version=1.18 --kernel_version=net-next`` will run specified test in 1.18 Kubernetes cluster running on net-next nodes. Kubernetes version defaults to 1.17, kernel version defaults to 4.19.
```

Comment:
So `retest-net-next` would be the same as `test-only --kernel_version=netnext --k8s_version=1.11 --focus="K8s"`?

Is there a way to test across kernel versions (e.g., `--kernel_version=all`) as what somewhat possible with `test-focus`?

Reply:
@pchaigno yes to first question, but it won't make the required CI check green. The answer to second question is No, would you say that there is need for this functionality?

Reply:
> The answer to second question is No, would you say that there is need for this functionality?

I wasn't mostly thinking that, if we had `kernel_version=all` and we defaulted to that, we could just type `test-only --focus="K8sSomeTestThatMayNotRunOnAllKernels"` without having to check on what kernel version the test is supported. But thinking about it again, maybe it's better if we have to explicitly set the exact config. we want; that should lead to less unnecessary tests.